### PR TITLE
UIU-1355: Users App | Add a Departments filter

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Add a Departments filter to filter users by their department

## Issue
[UIU-1355](https://issues.folio.org/browse/UIU-1355)

### CODE REVIEW

**Review by Vladyslav-Velytskyi — APPROVED** [review]

File: src/views/UserSearch/Filters.js [hunk 1]
```diff
@@ -94,27 +105,45 @@ export default class Filters extends React.Component {
...
           id="users-filter-accordion-patron-group"
           header={FilterAccordionHeader}
-          label={<FormattedMessage id="ui-users.information.patronGroup" />}
+          label={formatMessage({ id: 'ui-users.information.patronGroup' })}
           separator={false}
           onClearFilter={() => clearGroup('pg')}
         >
           <CheckboxFilter
-            dataOptions={this.getValuesFromResources('patronGroups', 'group')}
+            dataOptions={this.getValuesFromResources('patronGroups', 'group', 'group')}
```
Comment (mkuklis) [inline src/views/UserSearch/Filters.js:113]: @BogdanDenis would you mind changing this to `this.getValuesFromResources('patronGroups', 'group', 'id')`

Comment (zburke) [reply src/views/UserSearch/Filters.js:113]: This will restore the behavior of #1258 that was inadvertently removed in #1441.

**Review by zburke — APPROVED** [review]
This is a nicely implemented feature. Thanks for the additional comments around `getValuesFromResources()`!

**Review by mkuklis — APPROVED** [review]
Nicely done @BogdanDenis!

**Pull request merged**